pkg/codegen/nodejs/gen.go

```diff
@@ -42,6 +42,11 @@ import (
 	"github.com/pulumi/pulumi/sdk/v3/go/common/util/contract"
 )
 
+// The minimum version of @pulumi/pulumi compatible with the generated SDK.
+const MinimumValidSDKVersion string = "^3.42.0"
+const MinimumTypescriptVersion string = "^4.3.5"
+const MinimumNodeTypesVersion string = "^14"
```

Comment:
```suggestion
const MinimumNodeTypesVersion string = ">=14"
```

So users can select the appropriate version for them.